### ISSUE TITLE
`IPAddressToInterface` REST API Endpoint

### ISSUE THREAD

**Issue by pszulczewski** [nautobot/nautobot#4249]
### As ...

Tammy - Tools Integrator

### I want ...

In `2.0.0.b2` `Interface` to `IpAaddress` association has been changed to M2M. According to Django REST Framework docs M2M are read-only can't be set with respective model calls.

I talked to @glennmatthews and he suggests implementing custom API endpoint to allow `Interface` to 'IpAddress` association.

### So that ...

`nautobot-ansible` collection will have the same functionality with `2.0`.

### I know this is done when...

I can make this association via API using pynautobot (`next-2.0` branch)

### Optional - Feature groups this request pertains to.

- [ ] Automation
- [ ] Circuits
- [X] DCIM
- [X] IPAM
- [ ] Misc (including Data Sources)
- [ ] Organization
- [ ] Plugins (and other Extensibility)
- [ ] Security (Secrets, etc)
- [ ] Image Management
- [ ] UI/UX
- [ ] Documentation
- [ ] Other (not directly a platform feature)

### TODO

- [ ] Expose `IPAddressToInterface` through table in an api view
  - [ ] Add serializer
  - [ ] Add viewset
  - [ ] Add routes
  - [ ] Add filters
  - [ ] Add tests

**Comment by bryanculver** [comment]
@nautobot/core will discuss in design session then point

**Comment by gsnider2195** [comment]
### Option 1
Expose through table in an api view (undesirable by itself but required for implementation of option 2)

### Option 2 (breaking, must be 2.0.0)
Similar to Tags (writable nested representation)

How do we expose through table attributes (`is_primary`, `is_standby`, etc)

Example interface representation:

```json
    "ip_address_assignments": [
        {
            "id": "<uuid>",
            "object_type": "ipam.ipaddresstointerface",
            "url": "https://next.demo.nautobot.com/api/ipam/ipaddresstointerface/<uuid>/",
            "is_source": false,
            "is_destination": false,
            "is_default": false,
            "is_preferred": false,
            "is_primary": false,
            "is_secondary": false,
            "is_standby": false,
            "ip_address": {
                "id": "5af6e025-f6bf-4325-a247-3d61b6deca7e",
                "object_type": "ipam.ipaddress",
                "url": "https://next.demo.nautobot.com/api/ipam/ip-addresses/5af6e025-f6bf-4325-a247-3d61b6deca7e/"
            }
            // make sure to exclude the interface and vm_interface fields for this nested serializer
        }
    ],
```

### Option 3 (desirable, non-breaking, could be 2.0.x or higher in a separate issue)
Separate API endpoints (similar to m2m ORM API `.remove()`, `.add()`) (`/api/dcim/interfaces/<uuid>/add_ipaddress`)

How do we expose through table attributes (`is_primary`, `is_standby`, etc)

Is this really necessary with Option 2? Do we want to have 2 different ways to do the same thing?



## TODO

This issue becomes option 1, open follow-up issue for option 2 (blocked by this issue), and another issue for option 3 to be groomed later
